src/librustdoc/visit_ast.rs

```diff
@@ -257,7 +210,13 @@ impl<'a, 'tcx> RustdocVisitor<'a, 'tcx> {
         debug!("visiting item {:?}", item);
         let name = renamed.unwrap_or(item.ident.name);
 
-        if item.vis.node.is_pub() {
+        let is_pub = if let hir::ItemKind::Macro { is_exported: true, .. } = item.kind {
```

Comment:
This line makes me wonder if we could replace `is_exported` by making `item.vis` public.

Reply:
Isn't it already public? This used `item.vis.node.is_pub()` before - @inquisitivecrystal why did you change it?

Reply:
The problem is that a `#[macro_export] macro_rules!` item doesn't have a `vis.node` that shows it's public, but is anyway. petrochenkov suggested that I could replace situations like this with calls to `tcx.visibility()`, which seems work fine.